### PR TITLE
Add SVGUse as wrapper for svgs

### PR DESCRIPTION
**Pull request by magdalenaxm**
Adds the `SVGUse` component as wrapper for svg's to use the correct path in preview

### CODE REVIEW

File: site/src/common/helpers/SvgUse.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    height: number;
+}
+
+export const SVGUse: React.FunctionComponent<SVGUseProps> = ({ xlinkHref, width, height }) => (
+    <svg width={width} height={height}>
+        <use xlinkHref={createAssetUrl(xlinkHref)} />
+    </svg>
+);
+
+export function createAssetUrl(url: string): string {
```
[the remark this answers is not in the document]
Comment (magdalenaxm) [reply site/src/common/helpers/SvgUse.tsx:15]: No its not needed

File: site/src/common/helpers/SvgUse.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import * as React from "react";
+
+interface SvgUseProps {
+    xlinkHref: string;
+    width: number;
+    height: number;
+}
```
Comment (nsams) [inline site/src/common/helpers/SvgUse.tsx:7]: you could use React.HTMLProps<HTMLSvgElement> insteand of just width/height

**Review by johnnyomair — CHANGES_REQUESTED** [review]
Please change your example to include the Comet DXP logo in the header. We should be very careful about what we're adding to the Starter, as it will be the starting point for all our projects. Devs will probably remove the chevron from the text link block, but forget to delete the asset as well. If the asset would be the logo, then it would be more clear that it should be removed if it's not needed.

**Comment by magdalenaxm** [comment]
> Please change your example to include the Comet DXP logo in the header. We should be very careful about what we're adding to the Starter, as it will be the starting point for all our projects. Devs will probably remove the chevron from the text link block, but forget to delete the asset as well. If the asset would be the logo, then it would be more clear that it should be removed if it's not needed.

I have changed the logo now but when using the comet logo which i took from the comet repository, the svg is not showing in the preview anymore. Do I have to adapt the svg code or something?

**Comment by johnnyomair** [comment]
@magdalenaxm it seems to me that SVG use doesn't work when there are `defs` in the SVG. May be could simplify the logo to

```svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 133 34" id="logo">
    <path d="M8.21290465,9.18348765 L8.61237391,9.19099159 C9.66746566,9.2310109 10.6412716,9.43108711 11.5335744,9.79089506 L11.5335744,9.79089506 L11.8625798,9.933186 C12.7253831,10.3318549 13.4718174,10.8843621 14.101557,11.591358 L14.101557,11.591358 L12.6625279,12.9830247 L12.4537649,12.7736806 C11.3244374,11.691004 9.94031265,11.149537 8.30151977,11.149537 L8.30151977,11.149537 L7.95947485,11.1572684 C6.94330676,11.2036559 6.01684267,11.4587809 5.18019535,11.9225309 L5.18019535,11.9225309 L4.90730765,12.0830918 C4.10690855,12.5822932 3.47089791,13.2391821 2.99961395,14.054321 L2.99961395,14.054321 L2.8502483,14.3292417 C2.42575452,15.163625 2.21367674,16.0845833 2.21367674,17.0916667 L2.21367674,17.0916667 L2.22153194,17.4242498 C2.26866451,18.4123873 2.52791244,19.3140123 2.99961395,20.1290123 L2.99961395,20.1290123 L3.16280913,20.3948775 C3.67069484,21.1749275 4.34326965,21.7970525 5.18019535,22.2608025 L5.18019535,22.2608025 L5.46240418,22.4077381 C6.3190094,22.8253256 7.26541884,23.0339506 8.30151977,23.0339506 L8.30151977,23.0339506 L8.60339301,23.0275285 C10.1982638,22.9590264 11.5512228,22.34252 12.6625279,21.1783951 L12.6625279,21.1783951 L14.101557,22.5699074 L13.8597929,22.8279429 C13.2001278,23.4971346 12.4208081,24.0150892 11.5225942,24.3814815 L11.5225942,24.3814815 L11.1818287,24.5113347 C10.2633137,24.8372199 9.26633682,25 8.19078953,25 L8.19078953,25 L7.83587082,24.9939212 C6.42692622,24.9452909 5.14675501,24.6048789 3.99572209,23.9726852 L3.99572209,23.9726852 L3.68952836,23.7961384 C2.58742887,23.1291227 1.71202703,22.2456404 1.06260814,21.145216 L1.06260814,21.145216 L0.90540745,20.8655755 C0.301761938,19.7362864 3.68344363e-13,18.4782764 3.68344363e-13,17.0916667 L3.68344363e-13,17.0916667 L0.00628677742,16.7477011 C0.0565812784,15.3825836 0.408647853,14.1463794 1.06260814,13.0381173 L1.06260814,13.0381173 L1.23094111,12.7657613 C1.92364542,11.694607 2.84882907,10.8427113 4.00685698,10.2108025 C5.26107791,9.5257716 6.6633,9.18348765 8.21290465,9.18348765 L8.21290465,9.18348765 Z M36.5020047,9.18348765 L36.8376726,9.18969184 C37.1707259,9.20196759 37.4961023,9.23206019 37.8153023,9.27638889 L37.8153023,9.27638889 L36.1492453,11.1608025 L35.8199389,11.1844201 C34.9501757,11.2676002 34.1475782,11.5125171 33.4137756,11.9225309 L33.4137756,11.9225309 L33.1430077,12.0830918 C32.3486451,12.5822932 31.7157522,13.2391821 31.2443291,14.054321 L31.2443291,14.054321 L31.0950053,14.3292417 C30.6706243,15.163625 30.4585465,16.0845833 30.4585465,17.0916667 L30.4585465,17.0916667 L30.4664016,17.4242498 C30.5135327,18.4123873 30.7727667,19.3140123 31.2443291,20.1290123 L31.2443291,20.1290123 L31.4074505,20.3948775 C31.9147558,21.1749275 32.5835308,21.7970525 33.4137756,22.2608025 L33.4137756,22.2608025 L33.6936464,22.4077338 C34.5429054,22.8252963 35.4789872,23.0337963 36.5020047,23.0337963 L36.5020047,23.0337963 L36.8394094,23.0260733 C37.841375,22.979733 38.751081,22.7248302 39.567964,22.2608025 L39.567964,22.2608025 L39.8344632,22.100354 C40.6167138,21.6014151 41.243733,20.9441512 41.7152953,20.1290123 L41.7152953,20.1290123 L41.8725219,19.838033 C42.1198026,19.3470207 42.2933556,18.825823 42.3945233,18.2751543 L42.3945233,18.2751543 L44.6801128,16.3055556 L44.6992666,16.564369 C44.7095309,16.737963 44.7147547,16.9136831 44.7147547,17.0916667 L44.7147547,17.0916667 L44.7084705,17.4356399 C44.6581963,18.8008154 44.3062614,20.0373813 43.6523012,21.145216 L43.6523012,21.145216 L43.4839631,21.4176632 C42.79122,22.4891217 41.8657947,23.3406339 40.7080523,23.9726852 L40.7080523,23.9726852 L40.3913462,24.1367702 C39.2188207,24.7122567 37.9223338,25 36.5020047,25 L36.5020047,25 L36.1437258,24.9939212 C34.7216046,24.9452909 33.4314296,24.6048789 32.2736872,23.9726852 L32.2736872,23.9726852 L31.988993,23.8100317 C30.8693256,23.1408549 29.9789852,22.2489198 29.3184581,21.1341049 L29.3184581,21.1341049 L29.1595799,20.8529877 C28.5495079,19.7185788 28.2447151,18.464886 28.2447151,17.0916667 L28.2447151,17.0916667 L28.2510652,16.7509454 C28.3018674,15.3979683 28.6575029,14.1639008 29.3184581,13.0492284 L29.3184581,13.0492284 L29.4883774,12.7752287 C30.1872033,11.6977628 31.1158021,10.8427113 32.2736872,10.2108025 C33.5279081,9.52592593 34.9375535,9.18348765 36.5020047,9.18348765 L36.5020047,9.18348765 Z M62.5618081,9.36018519 L69.0257628,20.2285494 L75.4014116,9.36018519 L77.216707,9.36018519 L77.2388221,24.8231481 L75.1136058,24.8231481 L75.0916453,13.5572531 L69.4907988,22.9455247 L68.4724209,22.9455247 L62.8717291,13.6236111 L62.8717291,24.8231481 L60.7466674,24.8231481 L60.7466674,9.36018519 L62.5618081,9.36018519 Z M105.299657,9.36018519 L105.299657,11.2819444 L96.5776407,11.2819444 L96.5776407,16.0314815 L104.347779,16.0314815 L104.347779,17.9089506 L96.5776407,17.9089506 L96.5776407,22.9013889 L105.609578,22.9013889 L105.609578,24.8231481 L94.3638093,24.8231481 L94.3638093,9.36018519 L105.299657,9.36018519 Z M132.884785,9.36018519 L132.884785,11.2819444 L127.571899,11.2819444 L127.571899,24.8231481 L125.380337,24.8231481 L125.380337,11.2819444 L120.067451,11.2819444 L120.067451,9.36018519 L132.884785,9.36018519 Z M53.4980128,4.43533951 L42.4900977,16.720679 L42.4661493,16.4107746 C42.4039065,15.7968128 42.2556686,15.2179012 42.0228965,14.6734568 L42.0228965,14.6734568 L53.4980128,4.43533951 Z M53.991814,3.55271368e-15 L41.4950721,13.7032407 L41.3240637,13.4630408 C41.0301048,13.0718193 40.6915047,12.7260545 40.3095163,12.4242284 L40.3095163,12.4242284 L53.991814,3.55271368e-15 Z M47.0775151,2.95679012 L39.3483593,11.8074074 L39.0674999,11.6730581 C38.5937089,11.4619813 38.0914578,11.3150977 37.5607465,11.2324074 L37.5607465,11.2324074 L47.0775151,2.95679012 Z"/>            
</svg>
```

Oh, and please move the logo into the `Header` component 😁

File: site/src/common/helpers/SvgUse.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+interface SvgUseProps extends React.SVGProps<SVGSVGElement> {
+    xlinkHref: string;
+}
+
+export const SvgUse: React.FunctionComponent<SvgUseProps> = ({ xlinkHref, ...props }) => (
+    <svg {...props}>
+        <use xlinkHref={createAssetUrl(xlinkHref)} />
+    </svg>
+);
```
Comment (thomasdax98) [inline site/src/common/helpers/SvgUse.tsx:11]: Why are we using xlinkHref although it's deprecated (https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/xlink:href)? 
Support of xlinkHref is only ~ 0.7% better (https://caniuse.com/?search=use%20href).

MDN states
> If you need to support earlier browser versions, the deprecated xlink:href attribute can be used as a fallback in addition to the href attribute, e.g. <use href="some-id" xlink:href="some-id" x="5" y="5" />.

So we could do

```suggestion
interface SvgUseProps extends React.SVGProps<SVGSVGElement> {
    href: string;
}

export const SvgUse: React.FunctionComponent<SvgUseProps> = ({ href, ...props }) => {
    const assetUrl = createAssetUrl(href);
    return (
        <svg {...props}>
            <use href={assetUrl} xlinkHref={assetUrl} />
        </svg>
    );
};
```

File: site/src/layout/header/Header.tsx [hunk 4]
```diff
@@ -12,6 +13,7 @@ interface Props {
 function Header({ header }: Props): JSX.Element {
     return (
         <Root>
+            <SvgUse xlinkHref="/comet-logo.svg#logo" />
```
Comment (thomasdax98) [inline site/src/layout/header/Header.tsx:16]:
```suggestion
            <SvgUse href="/comet-logo.svg#logo" />
```

File: site/src/common/helpers/SvgUse.tsx (hunk 3 above)
Comment (johnnyomair) [reply site/src/common/helpers/SvgUse.tsx:11]: Nice catch! 👏🏼 We should use your suggestion.

**Review by johnnyomair — CHANGES_REQUESTED** [review]
Please implement the solution provided by @thomasdax98

**Review by thomasdax98 — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**